src/db-utils.js

```diff
@@ -38,11 +38,20 @@ export function recordWeight(username, weight) {
 export async function weightStats(username) {
     return new Promise((res) => {
         const db = new sqlite.Database(DB_LOCATION);
+        
         db.serialize(() => {
-            db.all(SELECT_USER_BIWEEKLY_AVERAGE, username, (err, rows) => {
-                res(rows.map((row) => row.weight));
-                db.close();
+            const data = new Object()
+            db.all(SELECT_CURRENT_WEEK, username, (err, rows) => {
```

Comment:
`db.all` will have it's callback run async IIRC. While this will not usually be the case due to order of queueing, it is possible that the second `db.all` runs to completion first and `data['currWeights']` may thus be missing. We also want to make sure `db.close()` is called after both callbacks have run to completion to avoid closing the connection prematurely.

We can wrap both queries in promises and use `Promise.all` to wait on the results and close the db connection if that sounds good.